### PR TITLE
Configure the project to run with Docker

### PR DESCRIPTION
**Pull request by kleberngueriero**
Add Dockerfile and docker-compose.yml to run the project and makes the configuration changes to run the project in Docker.
Also, add a docker-compose.override.yml that makes it easier to use in development.

Relative to issue https://github.com/call4paperz/call4paperz/issues/185

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM ruby:2.3.1
```
Comment (lucasrenan) [inline Dockerfile]: dumb question, which image is this one? which os?

**Comment by lucasrenan** [comment]
@kleberngueriero please add the issue in the PR description: https://github.com/call4paperz/call4paperz/issues/185

File: Dockerfile (hunk 1 above)
Comment (kleberngueriero) [inline Dockerfile]: I knew it was a debian or debian based, but just checked: It's Debian GNU/Linux 8(jessie).

**Comment by lucianosousa** [comment]
What about use `letter_opener` instead of `mailcatcher` ?
What do u guys think?

Less config, always better.

**Comment by lucasrenan** [comment]
@lucianosousa I'd say `letter_opener` will not work out with docker, as docker containers don't know how to launch a browser, which is is the host machine.

**Comment by kleberngueriero** [comment]
I believe would be ok to use letter_openner_web

**Comment by lucasrenan** [comment]
@kleberngueriero yeah, it may work with `letter_opener_web`. give it a try :)

**Comment by lucianosousa** [comment]
@jhonathas can you plz review this PR?
you PR was rejected with this idea in the past but I think we can't stop the `future` here.

I see with good eyes nowadays

**Comment by kleberngueriero** [comment]
@lucascaton and @lucianosousa, I commited the replacement of mailcatcher with letter_opener_web.
Also, since I was making changes:

- I renamed the docker-compose.override.yml to docker-compose.override.yml.sample, so anyone can edit it for personal use.
- I changed Dockerfile to make it easier to sync bundle installation in the host(avoiding redownload of all gems everytime one need to bundle install)

**Comment by lucascaton** [comment]
@kleberngueriero wrong Lucas 😄

cc: @lucasrenan

**Comment by kleberngueriero** [comment]
Oh man..sorry about that :sweat_smile:

**Comment by lucasrenan** [comment]
@lucascaton sorry man, it's always happening 😂

**Comment by lucascaton** [comment]
@lucasrenan no problem! I've contributed to this project before, it might be the reason why GitHub keeps suggesting my username in mentions 😬

**Comment by lucasrenan** [comment]
@jhonathas @lucianosousa any concerns regarding this PR? may we merge it?

**Comment by tauil** [comment]
I'm not a 100% aware about it, but as far as I know about Docker, it's fine to me. 👍

**Comment by lucianosousa** [comment]
@kleberngueriero merged. thanks!

**Comment by kleberngueriero** [comment]
@lucianosousa, should I delete the branch?

**Comment by lucasrenan** [comment]
@kleberngueriero go for it!

**Comment by lucianosousa** [comment]
@kleberngueriero I think I already did it here

**Pull request merged**